internal/caravan/tradeagreement.lua

```diff
@@ -26,13 +86,71 @@ local function diplomacy_toggle_cat()
     end
 end
 
+local function OrderGemWithMinValue(matPrices, val)
+    local priority_idx = GetCurrentTabId()
+
+    local priority = diplomacy.environment.dipev.sell_requests.priority[priority_idx]
+    for i in ipairs(priority) do
+        if matPrices[i] == val then
+            priority[i] = 4
+        end
+    end
+end
+
+local function GetCivMatPrices(tabSelected)
+    local resource = tabSelected.getCivResourceList(df.historical_entity.find(diplomacy.actor.civ_id).resources)
+    local matPrices = {}
+    local matValuesUnique = {}
+    local filter = {}
+    for civid, matid in pairs(resource) do
+        local matPrice = tabSelected.funcGetValue(matid)
+        matPrices[civid] = matPrice
+        if not filter[matPrice] then
+            local val = { value = matPrice, count = 1 }
+            filter[matPrice] = val
+            table.insert(matValuesUnique, val)
+        else
+            filter[matPrice].count = filter[matPrice].count + 1
+        end
+    end
+    table.sort(matValuesUnique, function(a, b) return a.value < b.value end)
+    return matPrices, matValuesUnique
+end
+
+local function ValueSelector()
```

Comment:
convention for local functions is snake_case, not PascalCase (which is generally used for class names)